biochatter/rag_agent.py

```diff
@@ -92,9 +99,14 @@ def __init__(
             self.agent.connect()
 
             self.query_func = self.agent.similarity_search
+
+        elif self.mode == RagAgentModeEnum.API:
+            from .api_agent import APIAgent
+            llm = ChatOpenAI(model_name='gpt-4', temperature=0, openai_api_key=os.getenv("OPENAI_API_KEY"))
```

Comment:
Here, instead of creating `llm` here, we'd better to pass in `conversation_factory` to `APIAgent`.

Reply:
implemented